database/migrations/2024_01_03_145604_create_enrollments_table.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateEnrollmentsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('enrollments', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id');
+            $table->unsignedBigInteger('course_id');
+            $table->string('provider_id')->nullable();
+            $table->enum('enrollment_state', ['active', 'inactive', 'completed', 'deleted']);
+            $table->json('links');
+            $table->timestamp('provider_start_at')->nullable();
+            $table->timestamp('provider_end_at')->nullable();
+            $table->timestamps();
+
+            $table->primary('id');
```

Comment:
Id is automatically marked primary, so this line is causing issues and you can remove it.